platform_buckets_api/kube_client.py

```diff
@@ -256,6 +260,11 @@ async def create_http_client(self) -> aiohttp.ClientSession:
             trace_configs=self._trace_configs,
         )
 
+    async def _reload_http_client(self) -> None:
+        await self.close()
+        self._token = None
```

Comment:
Will k8s automatically rotate token if we use legacy approach (service account secret mounting)?
Should we use [this approach](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#launch-a-pod-using-service-account-token-projection) instead?

Reply:
```
When a volume contains data from a Secret, and that Secret is updated, Kubernetes tracks this and updates the data in the volume, using an eventually-consistent approach.

Note: A container using a Secret as a [subPath](https://kubernetes.io/docs/concepts/storage/volumes#using-subpath) volume mount does not receive automated Secret updates.
```

https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets-as-files-from-a-pod

Reply:
same should apply to the SA token and certs.

Reply:
> However, the way you integrate with them has changed. Whereas there was a single legacy token per service account, always accessible at /var/run/secrets/kubernetes.io/serviceaccount/token, each pod may have multiple bound tokens. Because the tokens expire and are refreshed by Kubelet, applications need to periodically reload them from the filesystem.

Reply:
In AWS token expires every hour but for now AWS still accepts expired token for 90 days to give developers time to fix code. There is no guarantee that k8s syncs volumes every hour

Reply:
It would have been extremely unreliable if the legacy approach would not guarantee that the SA token is up-to-date. I agree though that the suggested approach is more robust.

Reply:
This is a good point, thank you.
I've checked our current clusters - all of them are injecting SA token projection into pods, even if SA has a linked secret with the token. AFAIU, the SA tokens populated via secrets by k8s are long-lived and never expire.

Or you are suggesting enforcing our deployments to use SA token projection?

Reply:
So your is ok, the only change we have to do is to add projection to k8s resource which will rotate token proactively.

Reply:
done in [9340f21](https://github.com/neuro-inc/platform-buckets-api/pull/407/commits/9340f215697c43e0540ed80ebcaef96082f1de7c)